Detectors/TRD/calibration/src/DCSProcessor.cxx

```diff
@@ -0,0 +1,246 @@
+        int32_t runNumber = o2::dcs::getValue<int32_t>(dpcom);
+        if (mVerbose) {
+          LOG(info) << "Run number = " << runNumber;
+        }
+        // end processing current DP, when it is of type trd_runNo
+      } else if (std::strstr(dpid.get_alias(), "trd_runType") != nullptr) { // DP is trd_runType
+        std::string aliasStr(dpid.get_alias());
+        int32_t runType = o2::dcs::getValue<int32_t>(dpcom);
+        if (mVerbose) {
+          LOG(info) << "Run type = " << runType;
+        }
+      }
+    }
+  }
+  return 0;
+}
+
+//______________________________________________________________________
+
+int DCSProcessor::processFlags(const uint64_t flags, const char* pid)
+{
+
+  // function to process the flag. the return code zero means that all is fine.
+  // anything else means that there was an issue
+
+  // for now, I don't know how to use the flags, so I do nothing
+
+  if (flags & DataPointValue::KEEP_ALIVE_FLAG) {
+    LOG(info) << "KEEP_ALIVE_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::END_FLAG) {
+    LOG(info) << "END_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::FBI_FLAG) {
+    LOG(info) << "FBI_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::NEW_FLAG) {
+    LOG(info) << "NEW_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::DIRTY_FLAG) {
+    LOG(info) << "DIRTY_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::TURN_FLAG) {
+    LOG(info) << "TURN_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::WRITE_FLAG) {
+    LOG(info) << "WRITE_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::READ_FLAG) {
+    LOG(info) << "READ_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::OVERWRITE_FLAG) {
+    LOG(info) << "OVERWRITE_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::VICTIM_FLAG) {
+    LOG(info) << "VICTIM_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::DIM_ERROR_FLAG) {
+    LOG(info) << "DIM_ERROR_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_DPID_FLAG) {
+    LOG(info) << "BAD_DPID_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_FLAGS_FLAG) {
+    LOG(info) << "BAD_FLAGS_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_TIMESTAMP_FLAG) {
+    LOG(info) << "BAD_TIMESTAMP_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_PAYLOAD_FLAG) {
+    LOG(info) << "BAD_PAYLOAD_FLAG active for DP " << pid;
+  }
+  if (flags & DataPointValue::BAD_FBI_FLAG) {
+    LOG(info) << "BAD_FBI_FLAG active for DP " << pid;
+  }
+
+  return 0;
+}
+
+//______________________________________________________________________
+
+void DCSProcessor::updateDPsCCDB()
+{
+  // here we create the object to then be sent to CCDB
+  LOG(info) << "Finalizing";
+  // TODO I do not know why this converter works, but I don't want to go deeper into the DCS code at the moment
+  union Converter {
```

Comment:
Hmm, this converter might be problematic. As I wrote elsewhere, I am not sure why we would get double instead of float.
I would have looked in the direction of a reinterpret_cast.
I am also wondering if endianness could be a problem? How is that on apple silicon / M1?

Reply:
It is not so clear to me, as in the DCS code the value is encapsulated in a 64 bit variable with a 56 bit payload (if I understand the code in `Detectors/DCS/include/DetectorsDCS/DataPointValue.h` correctly). The other detectors use the same conversion to get the payload from the data point. So I for now just copied that. Probably I need to ask someone from DCS about it, but I don't know whom.

Reply:
This is changed now and uses the `getValue` method provided from DCS